src/register/register.ts

```diff
@@ -52,6 +52,27 @@ export class Register {
       clipboard.copy(content);
     }
 
+    // Upper-case register name denotes append.
+    if (/^[A-Z]$/.test(register)) {
+     register = register.toLowerCase();
+     let existingText = Register.registers[register] && Register.registers[register].text;
+     if (existingText) {
+      if (typeof content === 'string') {
+        if (typeof existingText === 'string') {
+          content = existingText + (existingText.endsWith('\n') ? '' : '\n') + content;
+        } else {
+          content = existingText.join('\n') + '\n' + content;
```

Comment:
Careful. The reason that content can be a string[] is because of yanking in visual block mode - you copy an entire block, and then paste the whole block later. I'm don't think that flattening down that array is the right thing to do here, because then you lose that block.